### PR TITLE
Converts Libvcx Logging to Match Libindy's Logger

### PR DESCRIPTION
**Pull request by hadleym**
Mimics how libindy's logger operates,

Allows user to set default logger, set a custom logger, and retrieve the current logger.

Also implemented for python and node wrappers.

### CODE REVIEW

**Comment by hadleym** [comment]
@glowkey @ryMarsh44 
Gentlemen, please review.

**Comment by jovfer** [comment]
@Artemkaaas please review too.
I would like to have 2 reviews for this PR: one from core VCX team and from core libindy team

File: libindy/tests/ledger.rs [hunk 1]
```diff
@@ -673,7 +673,7 @@ mod high_cases {
                 },
             });
 
-            let request = ledger::build_get_schema_request(Some(IDENTIFIER), &anoncreds::gvt_schema_id()).unwrap();
+            let request = ledger::build_get_scuse utils::ctypes;hema_request(Some(IDENTIFIER), &anoncreds::gvt_schema_id()).unwrap();
```
[the remark this answers is not in the document]
Comment (hadleym) [reply libindy/tests/ledger.rs]: Agreed. Looks like some libindy files modifications slipped by me.

File: vcx/libvcx/include/vcx.h [hunk 2]
```diff
@@ -1577,6 +1578,33 @@ vcx_error_t vcx_wallet_validate_payment_address(int32_t command_handle,
                                              const char *payment_address,
                                              void (*cb)(int32_t, vcx_error_t));
 
+
+vcx_error_t vcx_set_default_logger( const char * pattern );
```
[the remark this answers is not in the document]
Comment (hadleym) [reply vcx/libvcx/include/vcx.h]: Typo fixed

File: vcx/libvcx/src/settings.rs [hunk 3]
```diff
@@ -187,9 +185,20 @@ pub fn test_agency_mode_enabled() -> bool {
 }
 
 pub fn process_config_string(config: &str) -> Result<u32, u32> {
+    use utils::logger::LibvcxDefaultLogger;
+    use serde_json::Value;
     let configuration: Value = serde_json::from_str(config).or(Err(error::INVALID_JSON.code_num))?;
     if let Value::Object(ref map) = configuration {
         for (key, value) in map {
+            if key == DEFAULT_LOGGING_KEY {
```
Comment (glowkey) [inline vcx/libvcx/src/settings.rs]: Can't we just use the RUST_LOG env variable for the default logger?  Seems easy and gets rid of this workaround.

Comment (hadleym) [reply vcx/libvcx/src/settings.rs]: yes, working on this change.

**Pull request merged**